clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -183,9 +183,9 @@ public void assignFromSubscribed(Collection<TopicPartition> assignments) {
         Set<TopicPartition> newAssignment = new HashSet<>(assignments);
         removeAllLagSensors(newAssignment);
 
-        for (TopicPartition tp : assignments)
-            if (!this.subscription.contains(tp.topic()))
-                throw new IllegalArgumentException("Assigned partition " + tp + " for non-subscribed topic.");
+//        for (TopicPartition tp : assignments)
```

Comment:
@hachikuji @mjsax Updated to use the `subscriptionPattern` if possible; this is discussed offline and please see if it looks "simple enough".